### ISSUE TITLE
instructions for windows users

### ISSUE THREAD

**Issue by smrgeoinfo** [linkml/linkml-template#22]
might be good to mention that 'make' is a linux command, and if you're working in Windows, installing Cygwin is very useful...

**Comment by hsolbrig** [comment]
Good input.  Out of curiosity, how much push-back to you think we'll get from Windows users in general?  We've been having a discussion about these sort of dependencies, but, as a lot of us are Mac users, we don't know how much trouble it will cause

**Comment by cmungall** [comment]
is it even necessary to be able to run locally? gh actions will do it

I think given current resources it is not a priority, but I think the
overall direction should be to allowing everyone to do everything via a web
browser

as an interim measure docker is an option too. we do this for odk, which is
very makefile centric?

**Comment by smrgeoinfo** [comment]
@cmungall I'm not sure what the GH actions are going to do- I don't see an actions in the template repo.    Doing everything in a browser is interesting.  How does that relate to what I see here?  Should I just go ahead and hand craft my JSON schema like I normally do and wait until the browser tools are working?  I'm trying this because the idea of creating the model once and being able to generate various useful representations is always attractive.  It just needs to be simple and the workflow clearly articulated, as well as the environment required to execute the workflow...

**Comment by cmungall** [comment]
yes we are getting ready to roll out new templates, I'll let you know later today what the timeline is

in the interim, it would be a shame to manually create the json-schema, since we can automate this

 1. is Docker an option for you?
 2. If not, how about just running the python code directly and ignoring the Makefile (all the Makefile does is run shell commands)

I am not a Windows person so my ability to help is limited. A while ago I tried to help someone who was trying to use Make with cygwin and it was painful due to differences in implementation. But if we ignore make, do you know if you are able to install the linkml python modules from the requirements.txt and run the generator commands?

**Comment by smrgeoinfo** [comment]
I've got the command line working and can generate the JSON schema via e.g. 
`pipenv run gen-json-schema iSamplesSchemaBasic.yml  > iSamplesSchemaBasic.json`


There are some problem in that but I'll put issues in the LinkML issue tracker.

thanks

**Comment by cmungall** [comment]
Awesome!

And would love to chat with you about iSamples some time
